superset/views/core.py

```diff
@@ -2290,10 +2288,15 @@ def sqllab_viz(self):
         SqlaTable = ConnectorRegistry.sources["table"]
         data = json.loads(request.form.get("data"))
         table_name = data.get("datasourceName")
-        table = db.session.query(SqlaTable).filter_by(table_name=table_name).first()
+        database_id = data.get("dbId")
+        table = (
+            db.session.query(SqlaTable)
+            .filter_by(database_id=database_id, table_name=table_name)
+            .one_or_none()
+        )
```

Comment:
@john-bodley regarding #5449 I'm assuming here that the unique constraint is `database_id, table_name`. It seemed strange to me having the `first` call on `table_name` only. Anyway does #5449 need a change here also?

Reply:
@dpgaspar I'm not overly familiar with the `/sqllab_viz/` route but assuming this has to do with creating a viz from a SQL Lab query I would think the it should probably contain the `database_id`. I gather schema is optional as it could be undefined in SQL Lab.